src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -75,8 +78,41 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
       workflowStore.toMap.get(id) foreach {_ ! SubscribeTransitionCallBack(sender())}
   }
 
+  /**
+   * Returns a `Future[Any]` which will be failed if there is no workflow with the specified id.
+   */
+  private def assertWorkflowExistence(id: WorkflowId): Future[Any] = {
+    // Confirm the workflow exists by querying its state.  If no state is found the workflow doesn't exist.
+    dataAccess.getWorkflowState(id) map {
+      case None => throw new WorkflowNotFoundException(s"Workflow '$id' not found")
+      case _ =>
+    }
+  }
+
+  private def assertCallExistence(id: WorkflowId, callFqn: FullyQualifiedName): Future[Any] = {
+    dataAccess.getExecutionStatus(id, callFqn) map {
+      case None => throw new CallNotFoundException(s"Call '$callFqn' not found in workflow '$id'.")
+      case _ =>
+    }
+  }
+
   private def workflowOutputs(id: WorkflowId): Future[binding.WorkflowOutputs] = {
-    dataAccess.getOutputs(id) map SymbolStoreEntry.toWorkflowOutputs
+    for {
+      _ <- assertWorkflowExistence(id)
+      outputs <- dataAccess.getOutputs(id)
+    } yield {
+      SymbolStoreEntry.toWorkflowOutputs(outputs)
+    }
+  }
+
+  private def callOutputs(workflowId: WorkflowId, callFqn: String): Future[binding.CallOutputs] = {
+    for {
+      _ <- assertWorkflowExistence(workflowId)
+      _ <- assertCallExistence(workflowId, callFqn)
```

Comment:
Oh I think I see... because the user might specify a bogus workflowId or callFqn?